src/main/java/com/redhat/devtools/lsp4ij/features/codeLens/UnresolvedCodeLensViewportContext.java

```diff
@@ -101,6 +104,14 @@ public void refreshCodeVision(@NotNull CodeLensDataResult result,
         }, VIEWPORT_CHANGE_DELAY_MS);
     }
 
+    private synchronized Alarm getScrollStopAlarm() {

```

Comment:
You might consider using double-checked locking here so as not to have to synchronize the entire method every time.

Reply:
I don't block every time, I call this method only if  scrollStopAlarm is null

Reply:
No, the entire method is `synchronized`, so a lock will be taken out on every invocation. You could instead do the following:

```java
    private volatile Alarm scrollStopAlarm = null;

    private Alarm getScrollStopAlarm() {
        if (scrollStopAlarm == null) {
            synchronized (this) {
                if (scrollStopAlarm == null) {
                    scrollStopAlarm = new Alarm(Alarm.ThreadToUse.POOLED_THREAD, this);
                }
            }
        }
        return scrollStopAlarm;
    }
```

and that way you only pay the penalty of `synchronized` when initialization is actually needed. Note that `volatile` modifier that has been added to the field as that's needed for proper DCL ordering of events.

Reply:
Oh, I see what you mean. You're checking _externally_ whether or not it's `null`. The code above will do the same thing but encapsulated. And even if you don't do what I've shown above, you need to mark the field as `volatile` for the same reason because what you've implemented is effectively double-checked locking.

Reply:
I don't understand what you mean , call this method like this:

```
if (scrollStopAlarm == null) {
            scrollStopAlarm = getScrollStopAlarm();
        }
```

Reply:
@SCWells72 I use now your code, thanks for your suggestion.